### PR TITLE
Exercise retrieval updates

### PR DESCRIPTION
**Pull request by BethanyG**
- Added code and some error handling for retrieving the exercise file name from either `config.json` or via the passed-in slug name.
- Changed the SLUG_RE code to accept an underscore as part of a valid slug name

Decided to not change the code to accept more than one exercise file since we don't currently have any mulit-file exercises in either the concept nor practice categories.

### CODE REVIEW

File: representer/__init__.py [hunk 1]
```diff
@@ -55,11 +57,35 @@ def represent(slug: utils.Slug, input: utils.Directory, output: utils.Directory)
...
+            raise FileNotFoundError("No exercise file was found in config.json.")
+
+    else:
+        src = f'{input.joinpath(slug.replace("-", "_"))}.py'
+
+        try:
+            src.read_text()
+        except AttributeError as err:
+            print('No exercise file was found in the input directory.', err)
+            return
```
Comment (cmccandless) [inline representer/__init__.py:82]:
```suggestion
        if not src.is_file():
            print('No exercise file was found in the input directory.', err)
            return
```

File: representer/utils.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 Slug = NewType("Slug", str)
 
-SLUG_RE = re.compile(r"^[a-z]+(-[a-z]+)*$")
+SLUG_RE = re.compile(r"^[a-z]+([-_][a-z]+)*$")
```
Comment (cmccandless) [inline representer/utils.py:20]: Slugs aren't supposed to contain underscore, and I don't see anything in your other changes that required this change. Can you clarify why this change is necessary?

**Review by cmccandless — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1 +1,3 @@
-* @exercism/maintainers-admin
+* @cmccandless
+* @bethanyg
```
Comment (BethanyG) [inline .github/CODEOWNERS:2]:
```suggestion
* @cmccandless @bethanyg
```

File: representer/utils.py (hunk 2 above)
Comment (BethanyG) [reply representer/utils.py:20]: There was some ambiguity in concept exercise naming.  That's been resolved, and the new requirement is that all concept exercises be named for their stories and use `kebab case`.  So yes, the underscore should be removed -- thanks for catching that!

Comment (BethanyG) [inline representer/utils.py:20]:
```suggestion
SLUG_RE = re.compile(r"^[a-z]+(-[a-z]+)*$")
```

**Review by cmccandless — APPROVED** [review]

**Pull request merged**